pkg/utils/status.go

```diff
@@ -221,33 +223,46 @@ func getDaemonSetStatus(ctx context.Context, instance *csmv1.ContainerStorageMod
 func calculateState(ctx context.Context, instance *csmv1.ContainerStorageModule, r ReconcileCSM, newStatus *csmv1.ContainerStorageModuleStatus) (bool, error) {
 	log := logger.GetLogger(ctx)
 	running := false
-	controllerReplicas, controllerStatus, controllerErr := getDeploymentStatus(ctx, instance, r)
+	// TODO: Currently commented this block of code as the API used to get the latest deployment status is not working as expected
+	// TODO: Can be uncommented once this issues gets sorted out
+	/* controllerReplicas, controllerStatus, controllerErr := getDeploymentStatus(ctx, instance, r)
+	expected, nodeStatus, daemonSetErr := getDaemonSetStatus(ctx, instance, r)
 	newStatus.ControllerStatus = controllerStatus
+	newStatus.NodeStatus = nodeStatus */
 	expected, nodeStatus, daemonSetErr := getDaemonSetStatus(ctx, instance, r)
 	newStatus.NodeStatus = nodeStatus
+	controllerReplicas := newStatus.ControllerStatus.Desired
+	controllerStatus := newStatus.ControllerStatus
 
 	newStatus.State = constants.Failed
-	log.Infof("deployment controllerReplicas [%d]", controllerReplicas)
+	log.Infof("deployment controllerReplicas [%s]", controllerReplicas)
 	log.Infof("deployment controllerStatus.Available [%s]", controllerStatus.Available)
 
 	log.Infof("daemonset expected [%d]", expected)
 	log.Infof("daemonset nodeStatus.Available [%s]", nodeStatus.Available)
 
-	if (fmt.Sprintf("%d", controllerReplicas) == controllerStatus.Available) && (fmt.Sprintf("%d", expected) == nodeStatus.Available) {
+	if (controllerReplicas == controllerStatus.Available) && (fmt.Sprintf("%d", expected) == nodeStatus.Available) {
 		running = true
 		newStatus.State = constants.Succeeded
 	}
 	log.Infof("calculate overall state [%s]", newStatus.State)
-	var err error
-	if controllerErr != nil {
-		err = controllerErr
-	}
+	var err error = nil
+	// TODO: Uncomment this when the controller runtime API gets fixed
+	/*
+		if controllerErr != nil {
+			err = controllerErr
+		}
+		if daemonSetErr != nil {
+			err = daemonSetErr
+		}
+		if daemonSetErr != nil && controllerErr != nil {
+			err = fmt.Errorf("ControllerError: %s, Daemonseterror: %s", controllerErr.Error(), daemonSetErr.Error())
+			log.Infof("calculate overall error msg [%s]", err.Error())
+		} */
+
 	if daemonSetErr != nil {
 		err = daemonSetErr
-	}
-	if daemonSetErr != nil && controllerErr != nil {
-		err = fmt.Errorf("ControllerError: %s, Daemonseterror: %s", controllerErr.Error(), daemonSetErr.Error())
-		log.Infof("calculate overall error msg [%s]", err.Error())
+		log.Infof("calculate overall Daemonseterror msg [%s]", daemonSetErr.Error())
```

Comment:
Should  the log statement be Daemonset error msg?

Reply:
Addressed